src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -223,10 +223,12 @@ private DRSContentsObject makeDrsContentsObject(FSItem fsObject, String snapshot
 
     private String makeHttpsFromGs(String gspath) {
         try {
-            URI gsuri = URI.create(gspath);
-            String gsBucket = gsuri.getAuthority();
-            String gsPath = StringUtils.removeStart(gsuri.getPath(), "/");
-            String encodedPath = URLEncoder.encode(gsPath, StandardCharsets.UTF_8.toString());
+            String gsBucket = GcsPdao.extractBucketFromPath(gspath);
+            String gsPath = GcsPdao.extractFilePathInBucket(gspath, gsBucket);
+            String encodedPath = URLEncoder.encode(gsPath, StandardCharsets.UTF_8.toString())
+                // Google uses %20 instead of + to encode spaces.
+                // Original + characters should already have been converted to %2B
```

Comment:
These comments are confusing to me. If I understand correctly, there are `+` characters that exist that need to be converted to `%20`. Is this correct?

Reply:
That's right. Is a user has a file named, say `my file+.txt`, URLEncoder.encode` converts that to `my+file%2B.txt`.  The HTTPS access for GCS objects don't quite recognize that.  They require spaces to be converted to `%20`.  Now, at this point, you'll notice that I had a `+1` in there what was converted to `%2B` so any `+`'s that are there have to be from a space that's been converted to `+`, so it's safe to convert those plusses to `%2B`.  That's kinda hard to put into a comment :)

Reply:
How about something like "Google does not recognize the `+` characters that are produced from spaces by the `URLEncoder.encode` method. As a result, these must be converted to `%2B`."